src/LNClient.ts

```diff
@@ -0,0 +1,100 @@
+import { fiat } from "@getalby/lightning-tools";
+import {
+  NewNWCClientOptions,
+  Nip47MakeInvoiceRequest,
+  Nip47Notification,
+  Nip47PayInvoiceRequest,
+  Nip47Transaction,
+  NWCClient,
+} from "./NWCClient";
+
+type LNClientCredentials = string | NWCClient | NewNWCClientOptions;
+type FiatAmount = { amount: number; currency: string };
+
+/**
+ * An amount in satoshis, or an amount in a fiat currency
+ */
+type Amount = number | FiatAmount;
+
+// Most popular fiat currencies
+export const USD = (amount: number) =>
+  ({ amount, currency: "USD" }) satisfies FiatAmount;
+export const EUR = (amount: number) =>
+  ({ amount, currency: "EUR" }) satisfies FiatAmount;
+export const JPY = (amount: number) =>
+  ({ amount, currency: "JPY" }) satisfies FiatAmount;
+export const GBP = (amount: number) =>
+  ({ amount, currency: "GBP" }) satisfies FiatAmount;
+export const CHF = (amount: number) =>
+  ({ amount, currency: "CHF" }) satisfies FiatAmount;
+
+export class LNClient {
+  readonly nwcClient: NWCClient;
+
+  constructor(credentials: LNClientCredentials) {
+    if (typeof credentials === "string") {
+      this.nwcClient = new NWCClient({
+        nostrWalletConnectUrl: credentials,
+      });
+    } else if (credentials instanceof NWCClient) {
+      this.nwcClient = credentials;
+    } else {
+      this.nwcClient = new NWCClient(credentials);
+    }
+  }
+
+  pay(invoice: string, args?: Omit<Nip47PayInvoiceRequest, "invoice">) {
+    return this.nwcClient.payInvoice({
+      invoice,
+      ...(args || {}),
```

Comment:
amount needs to be fixed here or removed (sat - millisat issue - and only required for 0-amount invoices)

Reply:
EDIT: also for lightning addresses, I need to update this